django/core/validators.py

```diff
@@ -282,7 +282,7 @@ def validate_ipv4_address(value):
         # Leading zeros are forbidden to avoid ambiguity with the octal
         # notation. This restriction is included in Python 3.9.5+.
         # TODO: Remove when dropping support for PY39.
-        if any(octet != "0" and octet[0] == "0" for octet in value.split(".")):
+        if any(octet != "0" and octet[0] == "0" for octet in str(value).split(".")):
```

Comment:
To be consistent with previous fixes, wouldn't it be better to return early if `isinstance(value, ipaddress.IPv4Address)`?